### ISSUE TITLE
Nested directories not supported

### ISSUE THREAD

**Issue by joneskoo** [gokrazy/internal#6]
As discussed on twitter, I'd like to use this to build alpine-pkr, a disk image builder for headless Alpine Linux Raspberry images.

I'm making some progress but I suspect gokrazy/internal/fat creates corrupted FAT in case of nested directories. If I strip /apks it will generate a valid FAT according to dosfsck (and no longer hangs macOS when mounting the image :D ).

```
fsck.fat 4.1 (2017-01-24)
/apks/aarch64
 Start does point to root directory. Deleting dir. 
Reclaimed 4966 unused clusters (10170368 bytes).
Perform changes ? (y/n) ^C
```

**Comment by joneskoo** [comment]
I wonder if this should set firstCluster to non-zero for nested directories?

https://github.com/gokrazy/internal/blob/539bb61868dee732bbf57f3c62cb66aaf405ae38/fat/writer.go#L208-L210

I don't know where to start looking for the problem here…

**Comment by stapelberg** [comment]
Sorry, I don’t have much time to look into these details currently, as you can tell by my late reply :)

In general, I found https://kaitai.io/ helpful to dive into FAT file systems.

You could try using mkfs.vfat to generate a small file system, then just do the minimum modifications you need, and inspect the resulting file system manually and see if you can spot any differences.

**Comment by stapelberg** [comment]
There is a good chance this was fixed with commit f2bd9b32d81621684797e7e56bf4078581298e59

Please re-open if you run into issues. One currently un-implemented corner case is having lots of files in the same directory.
